### ISSUE TITLE
Invalid suggestion on missing lifetime in impl trait associated type

### ISSUE THREAD

**Issue by pustaczek** [rust-lang/rust#68904]
Trying to elide lifetimes in an associated type of an impl trait causes rustc to emit some suggestions that contain invalid Rust syntax.

**I tried this code:**

```rust
trait Traito {
	type Typeo;
}

fn f(_: impl Traito<Typeo=&()>) {
}

```

**I expected to see this happen:** an error is emitted, possibly with a suggestion of adding a generic lifetime parameter.

**Instead, this happened:** the suggestion contains invalid Rust syntax, where the lifetime appears as if the parameter type was the generic parameter list:

```
    Checking wibblywobbly v0.1.0 (/home/matcegla/wibblywobbly)
error[E0106]: missing lifetime specifier
 --> src/lib.rs:5:27
  |
5 | fn f(_: impl Traito<Typeo=&()>) {
  |                           ^ expected named lifetime parameter
  |
help: consider introducing a named lifetime parameter
  |
5 | fn f(_: 'lifetime, impl Traito<Typeo=&'lifetime ()>) {
  |         ^^^^^^^^^^                   ^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0106`.
error: could not compile `wibblywobbly`.

To learn more, run the command again with --verbose.
```

*rustc --version --verbose:
rustc 1.43.0-nightly (58b834344 2020-02-05)
binary: rustc
commit-hash: 58b834344fc7b9185e7a50db1ff24e5eb07dae5e
commit-date: 2020-02-05
host: x86_64-unknown-linux-gnu
release: 1.43.0-nightly
LLVM version: 9.0*

**Comment by estebank** [comment]
This was fixed in #68583: https://github.com/rust-lang/rust/pull/68583/files#diff-d4a8232a0918ba144fa7f17e2d941851.

It was very recently merged so it should be visible in tomorrow's nightly, I think.
